pkg/frontend/querymiddleware/astmapper/instant_splitting_test.go

```diff
@@ -140,13 +140,13 @@ func TestInstantSplitter(t *testing.T) {
 		// Should map only left-hand side operand of inner binary operation, if right-hand side range interval is too small
 		{
 			in:                   `sum(sum_over_time({app="foo"}[3m]) + count_over_time({app="foo"}[1m]))`,
-			out:                  `sum ((sum without() (` + concatOffsets(splitInterval, 3, `sum_over_time({app="foo"}[x]y)`) + `)) + (count_over_time({app="foo"}[1m])))`,
+			out:                  `sum ((sum without() (` + concatOffsets(splitInterval, 3, `sum_over_time({app="foo"}[x]y)`) + `)) + ` + concat(`(count_over_time({app="foo"}[1m]))`) + `)`,
```

Comment:
Note to reviewers: The `concat()` makes the input query(ies) being pushed down. That's required every time we have a vector selector, otherwise the query-frontend won't be able to fetch the samples (it must be pushed down to queriers).

Reply:
Hum, interesting 🤔 so this means that for every binary operation that at least one operand is mapped, we need to wrap the non-split operand in concat?
Maybe we should also add a test of a binary expression not included in a sum? e.g., `sum_over_time({app="foo"}[3m]) * count_over_time({app="foo"}[1m])`

Reply:
Small nit ^ adding a binary expression not included in the sum where one of the legs is not mapped:
`sum_over_time({app="foo"}[3m]) * count_over_time({app="foo"}[1m])`

Reply:
> so this means that for every binary operation that at least one operand is mapped, we need to wrap the non-split operand in concat?

Yes. It's just a trick to push down the non-split leg too, otherwise it won't be executed by a querier but the query-frontend can't execute it (because it's not capable of fetching samples from ingesters / store-gateways).

Reply:
> Maybe we should also add a test of a binary expression not included in a sum?

Sorry, I totally missed these comments. Addressed.